src/Symfony/Component/Security/Http/Authenticator/AccessToken/BodyAccessTokenExtractor.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Extracts a token from the body request.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ *
+ * @final
+ */
+class BodyAccessTokenExtractor implements AccessTokenExtractorInterface
+{
+    public const PARAMETER = 'access_token';
+    private string $parameter;
+
+    public function __construct(string $parameter = self::PARAMETER)
+    {
+        $this->parameter = $parameter;
+    }
+
+    public function extractToken(Request $request): ?string
+    {
+        if (Request::METHOD_POST !== $request->getMethod()) {
+            return null;
+        }
+        if (!str_starts_with($request->headers->get('CONTENT_TYPE', ''), 'application/x-www-form-urlencoded')) {
+            return null;
+        }
+        $content = $request->getContent();
+        if (!\is_string($content)) {
+            return null;
+        }
+
+        parse_str($content, $body);
```

Comment:
use `$request->request` to read the form-encoded body content instead of reimplementing the body parsing.

Reply:
OK. Done in the last commit.